cobalt/webdriver/dispatcher_test.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/webdriver/dispatcher.h"
+
+#include <utility>
+
+#include "base/optional.h"
+#include "base/values.h"
+#include "cobalt/webdriver/protocol/window_id.h"
+#include "cobalt/webdriver/session_driver.h"
+#include "cobalt/webdriver/util/command_result.h"
+#include "cobalt/webdriver/util/dispatch_command_factory.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "url/gurl.h"
+
+namespace cobalt {
+namespace webdriver {
+
+TEST(FromValueTest, WindowID) {
+  base::Value::Dict dict;
+  dict.Set("name", "foo");
+  base::Value value(std::move(dict));
+  auto result = util::internal::FromValue<protocol::WindowId>(&value);
+  EXPECT_EQ(result->id(), "foo");
+}
+
+TEST(FromValueTest, WindowID_Fail) {
+  base::Value empty;
+  auto result = util::internal::FromValue<protocol::WindowId>(&empty);
+  EXPECT_EQ(result, base::nullopt);
```

Comment:
Should be comparing it to absl::nullopt

Reply:
I wrote this with base:: APIs to ensure i can cherrypick the same test to C24 and cross check. Do you want this to be changed ? We have extensive `base::nullopt` usage in a lot of code still.

Reply:
Either works and that's a fair reason but base::nullopt only exists for compatibility (I added it) and is deprecated (right now it's just a shim that points to absl::nullopt)

Reply:
I'll leave as is for now, and then we do wholesale nullopt replacement.